SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -6591,35 +6657,60 @@ void CEulerSolver::SetPrimitive_Limiter(CGeometry *geometry, CConfig *config) {
 
         /*--- Calculate the interface left gradient, delta- (dm) ---*/
         
-        dm = 0.0;
-        for (iDim = 0; iDim < nDim; iDim++)
-          dm += 0.5*(Coord_j[iDim]-Coord_i[iDim])*Gradient_i[iVar][iDim];
+        switch(nDim)
```

Comment:
Two reasons - first like above removing a very frequent called short loop. Next the precalculation of Coordiv (independent of the nVar-loop) and with a change in sign to be used for i and j limiter as well.